### PR TITLE
fix(auto-import-plugin): support functional components

### PR DESCRIPTION
**Pull request by mengqiuleo**
# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-vue/blob/dev/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #2402 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Enhanced documentation for integrating the `TinyVue` component library with Vite and Webpack, including new examples for functional components.
	- Added global constants for `TinyLoading`, `TinyModal`, and `TinyNotify` to simplify usage across the application.
	- Introduced new configuration options for automatic imports in Vite and Webpack.

- **Improvements**
	- Transitioned `App.vue` to use the Composition API for better state management and user interaction handling.
	- Updated Vite configuration to improve plugin integration and component auto-importing.

- **Dependency Updates**
	- Upgraded `@opentiny/vue` and `vite` dependencies, and added `unplugin-auto-import`.

### CODE REVIEW

File: examples/sites/demos/pc/webdoc/import-components.md [hunk 1]
```diff
@@ -73,7 +73,55 @@ module.exports = defineConfig({
   configureWebpack: {
     plugins: [
       Components({
-        resolvers: [TinyVueResolver]
+        resolvers: [TinyVueResolver()]
```
Comment (shenjunjian) [inline examples/sites/demos/pc/webdoc/import-components.md]: TinyVueResolver 从单层函数， 变成双层函数了。 对于已经使用该变量的项目来说， 是一个破坏性变更。

是不是可以通过新增一个导出变量，比如 TinyVueAutoImportor 的变量呢？

Comment (mengqiuleo) [reply examples/sites/demos/pc/webdoc/import-components.md]: 已修复，通过新增导出变量解决

File: internals/unplugin-tiny-vue/src/index.ts [hunk 2]
```diff
@@ -12,7 +12,25 @@ const supportMap = {
...
+
+  if (componentName.startsWith('Tiny') && !componentName.startsWith('TinyIcon')) {
+    return {
+      name: componentName.slice(4),
+      from: '@opentiny/vue'
+    }
+  }
+}
+
+export const TinyVueResolver = (componentName: string) => {
```
Comment (shenjunjian) [inline internals/unplugin-tiny-vue/src/index.ts]: 为了避免破坏性变更， 你是保留了 TinyVueResolver 的名字， 特意新加了一个TinyVueAutoImportResolver 变量。

但现在看来， TinyVueAutoImportResolver 变量的能力涵盖了 TinyVueResolver， 且都是一层函数了
建议：只导出 TinyVueResolver变量， 但它的能力是 上面函数的能力即可。

File: examples/sites/demos/pc/webdoc/import-components.md [hunk 3]
```diff
@@ -80,6 +80,54 @@ module.exports = defineConfig({
...
+// vite.config.js
+import Components from 'unplugin-vue-components/vite'
+import AutoImport from 'unplugin-auto-import/vite'
+import { TinyVueAutoImportResolver } from '@opentiny/unplugin-tiny-vue'
+
+module.exports = defineConfig({
+  configureWebpack: {
+    plugins: [
+      Components({
+        resolvers: [TinyVueAutoImportResolver]
```
Comment (shenjunjian) [inline examples/sites/demos/pc/webdoc/import-components.md]: 这两个变量，都还使用  TinyVueResolver 这个名称即可

Comment (mengqiuleo) [reply examples/sites/demos/pc/webdoc/import-components.md]: 已修复. PTAL

File: internals/unplugin-tiny-vue/example/vite.config.js [hunk 4]
```diff
@@ -1,11 +1,22 @@
 import { defineConfig } from 'vite'
 import vue from '@vitejs/plugin-vue'
 import Inspect from 'vite-plugin-inspect'
-import autoImportPlugin from '../dist/index.js'
+import Components from 'unplugin-vue-components/vite'
+import AutoImport from 'unplugin-auto-import/vite'
+import { TinyVueAutoImportResolver } from '../dist/index.js'
```
Comment (shenjunjian) [inline internals/unplugin-tiny-vue/example/vite.config.js:6]: 这里的名称又漏改为 TinyVueResolver了。

**Review by zzcr — APPROVED** [review]

**Pull request merged**